app/src/util/tools.ts

```diff
@@ -625,8 +625,11 @@ export const clampBigNumber = (x: BigNumber, min: BigNumber, max: BigNumber): Bi
   return x
 }
 
-export const numberToByte32 = (num: number): string => {
-  const hex = new BigNumber(num).toHexString()
+export const numberToByte32 = (num: string | number, isScalar: boolean): string => {
+  let hex: any
+  if (isScalar) hex = num
```

Comment:
Why is it different for scalar?

Reply:
Outcome index is a number while bondOutcomeSelected is bigNumber...Should i have handled it differently?